### PR TITLE
[FLINK-24351][docs] Translate "JSON Function" pages into Chinese

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
## What is the purpose of the change
This pull request allows JSON functions to be interpreted in Chinese. This makes it more friendly for engineers from China, as well as easier to use.

## Brief change log
 Change the English part in /data/sql_functions_zh.yml under flink project to Chinese.

## Verifying this change
This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no) no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no) no
  - The serializers: (yes / no / don't know) no
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know) no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)no
  - The S3 file system connector: (yes / no / don't know)no

## Documentation

  - Does this pull request introduce a new feature? (yes / no)no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)docs

### CODE REVIEW

File: docs/data/sql_functions_zh.yml [hunk 1]
```diff
@@ -765,10 +764,9 @@ json:
   - sql: JSON_STRING(value)
     table: jsonString(value)
     description: |
-      Serializes a value into JSON.
+      将值序列化为JSON。
 
-      This function returns a JSON string containing the serialized value. If the value is `NULL`,
-      the function returns `NULL`.
+      这个函数返回一个包含序列化值的JSON字符串。如果值为 `NULL`，函数返回 `NULL`。
```
Comment (MonsterChenzhuo) [inline docs/data/sql_functions_zh.yml:769]:
```suggestion
      此函数返回一个包含序列化值的JSON字符串。如果值为 `NULL`，函数返回 `NULL`。
```

File: docs/data/sql_functions_zh.yml [hunk 2]
```diff
@@ -956,13 +940,11 @@ json:
   - sql: JSON_ARRAYAGG(items [ { NULL | ABSENT } ON NULL ])
     table: jsonArrayAgg(JsonOnNull, itemExpression)
     description: |
-      Builds a JSON object string by aggregating items into an array.
+      通过将字段聚合到数组中构建JSON对象字符串。
 
-      Item expressions can be arbitrary, including other JSON functions. If a value is `NULL`, the
-      `ON NULL` behavior defines what to do. If omitted, `ABSENT ON NULL` is assumed by default.
+      项表达式可以是任意的，包括其他 JSON 函数。如果值为 `NULL`，则 `ON NULL` 行为定义了要执行的操作。如果省略，默认情况下假定为 `ABSENT ON NULL`。
```
Comment (MonsterChenzhuo) [inline docs/data/sql_functions_zh.yml:945]:
```suggestion
      项目表达式可以是任意的，包括其他 JSON 函数。如果值为 `NULL`，则 `ON NULL` 行为定义了要执行的操作。如果省略，默认情况下假定为 `ABSENT ON NULL`。
```

File: docs/data/sql_functions_zh.yml [hunk 3]
```diff
@@ -908,15 +897,14 @@ json:
...
-      arbitrary, including other JSON functions. If a value is `NULL`, the `ON NULL` behavior
-      defines what to do. If omitted, `NULL ON NULL` is assumed by default.
+      键表达式必须返回不为空的字符串。值表达式可以是任意的，包括其他 JSON 函数。
+      如果值为 `NULL`，则 `ON NULL` 行为定义了要执行的操作。如果省略，默认情况下假定为 `NULL ON NULL`。
 
-      Note that keys must be unique. If a key occurs multiple times, an error will be thrown.
+      请注意，键必须是唯一的。如果一个键出现多次，将抛出一个错误。
 
-      This function is currently not supported in `OVER` windows.
+      此函数目前在 `OVER` windows 中不受支持。
```
Comment (RocMarshal) [inline docs/data/sql_functions_zh.yml:907]:
```suggestion
      目前在 `OVER` windows 中不支持此函数。
```

**Comment by RocMarshal** [comment]
@MonsterChenzhuo Thanks for the update. Could you please have a check on the ci result?

**Comment by RocMarshal** [comment]
@MonsterChenzhuo Please also make sure that the PR template is properly filled in, which makes it useful for reviewers to comprehend the change details of the PR.

**Comment by MonsterChenzhuo** [comment]
@RocMarshal  Hello, the pr template and ci have been completed

**Review by RocMarshal — COMMENTED** [review]
Thanks @MonsterChenzhuo  for the update. 
LGTM +1.
Now ping @wuchong @klion26 .

**Review by gaoyunhaii — APPROVED** [review]
LGTM, thanks @MonsterChenzhuo for the PR and @RocMarshal for the review!

**Pull request closed without merging**